src/amplify-migration/jsDomUtils.ts

```diff
@@ -0,0 +1,186 @@
+          rawPermalink,
+          changedPermalinks[rawPermalink]
+        );
+        fileContent = fileContent.replace(
+          rawPermalink,
+          changedPermalinks[rawPermalink]
+        );
+      }
+
+      const { fileContent: href } =
+        await updateFilesUploadsPath(
+          a.href,
+          setOfAllDocumentsPath,
+          currentRepoName
+        );
+      if (a.href !== href) {
+        fileContent.replace(a.href, href)
+      }
+
+    } else if (attribute === "src") {
+      /**
+       * Ideally this code should be const img = tag as HTMLImageElement;
+       * However, HTMLImageElement is a built-in interface in the browser's DOM API,
+       * which means that it's only available in a browser environment.
+       * In development, hovering over tag will show the inferred type as HTMLImageElement
+       */
+      const img = tag as any;
+
+      let rawPermalink = getRawPermalink(img[attribute]);
+
+      if (changedPermalinks[rawPermalink]) {
+        img[attribute] = img.src.replace(
+          rawPermalink,
+          changedPermalinks[rawPermalink]
+        );
+      }
+
+      const { fileContent:src } =
+        await updateFilesUploadsPath(
+          img.src,
+          setOfAllDocumentsPath,
+          currentRepoName
+        );
+      if (img.src !== src) {
+        fileContent.replace(img.src, src)
+      }
+    }
+  }
+  return { fileContent, dom, changedPermalinks };
+}
+/**
+ * Requirements:
+ * convert `/some/path` to `some/path`
+ * convert `some/path/` to `some/path`
+ * above two variants with `permalink: ` prefix
+ * above two variants with `url: ` prefix
+ * @param permalink original permalink
+ * @returns raw permalinks without leading/trailing slash
+ */
+
+export function getRawPermalink(permalink: string) {
+  let trimmedPermalink = permalink.trim();
+  if (trimmedPermalink.startsWith(`permalink: `)) {
+    trimmedPermalink = permalink.trim().slice(11);
+  }
+  if (trimmedPermalink.startsWith(`url: `)) {
+    trimmedPermalink = permalink.trim().slice(5);
+  }
+  if (trimmedPermalink.startsWith(`/`)) {
+    trimmedPermalink = trimmedPermalink.slice(1);
+  }
+  if (trimmedPermalink.endsWith("/")) {
+    trimmedPermalink = trimmedPermalink.slice(0, -1);
+  }
+
+  return trimmedPermalink;
+}
+/**
+ * We need to modify trailing slashes from files. We get the list of added extensions from
+ * https://github.com/isomerpages/isomercms-backend/blob/develop/src/utils/file-upload-utils.js
+ */
+
+export async function updateFilesUploadsPath(
+  fileContent: string,
+  setOfAllDocumentsPath: Set<string>,
+  currentRepoName: string
+): Promise<{ fileContent: string }> {
+  const fileRegexWithTrailingSlash = /\/(files|images)\/.*.(pdf|png|jpg|gif|tif|bmp|ico|svg)\//g;
+  const matches = fileContent.match(fileRegexWithTrailingSlash);
+  if (matches) {
+    matches.forEach(async (match) => {
+      assert(match.endsWith("/")); // sanity check that should have been guaranteed by regex
```

Comment:
would this throw?

Reply:
nope, just a sanity check bah, shouldnt throw